tx-pool/src/pool.rs

```diff
@@ -465,6 +518,116 @@ impl TxPool {
         (entries, size, cycles)
     }
 
+    pub(crate) fn check_rbf(
+        &self,
+        snapshot: &Snapshot,
+        rtx: &ResolvedTransaction,
+        conflicts: &HashSet<ProposalShortId>,
+        fee: Capacity,
+        tx_size: usize,
+    ) -> Result<(), Reject> {
+        assert!(self.enable_rbf());
+        assert!(!conflicts.is_empty());
+
+        let short_id = rtx.transaction.proposal_short_id();
+        // Rule #4, new tx's fee need to higher than min_rbf_fee computed from the tx_pool configuration
+        // Rule #3, new tx's fee need to higher than conflicts, here we only check the root tx
+        if let Some(min_replace_fee) = self.calculate_min_replace_fee(conflicts, tx_size) {
+            if fee < min_replace_fee {
+                return Err(Reject::RBFRejected(format!(
+                    "Tx's current fee is {}, expect it to >= {} to replace old txs",
+                    fee, min_replace_fee,
+                )));
+            }
+        } else {
+            return Err(Reject::RBFRejected(
+                "calculate_min_replace_fee failed".to_string(),
+            ));
+        }
+
+        let pool_entries = conflicts
+            .iter()
+            .map(|id| {
+                self.get_pool_entry(id)
+                    .expect("conflict Tx should be in pool")
+            })
```

Comment:
To minimize the potential bug impact, I suggest using `filter_map` here instead of `map/expect`

Reply:
I think it's better to use `map/expect` since it's like an assertion, `self.get_pool_entry(id)` returning `None` implies an inconsistency(invalid id or something), even we skip the error the result may be wrong, is it is a worser situation.

Reply:
ah, let me reconsider it to confirm it won't impact other tx by checking the top-down path.

Reply:
fixed in https://github.com/nervosnetwork/ckb/pull/4079/commits/95e3596c3943ec84116e6d2be9146e927eb89c7b, thanks.